src/runtime/analytics-service.js

```diff
@@ -84,21 +96,62 @@ export class AnalyticsService {
     return this.iframe_;
   }
 
+  /**
+   * @return {string}
+   * @private
+   */
+  getQueryString_() {
+    return this.doc_.getWin().location.search.slice(1);
+  }
+
+  /**
+   * @return {string}
+   * @private
+   */
+  getReferrer_() {
+    return parseUrl(this.doc_.getWin().document.referrer).origin;
```

Comment:
I'd rather have `parse().origin` to be done outside of this class. This is something that we critically need to test and `getReferrer_` is here to mock out this part of code.

Reply:
done